molecule/__init__.py

```diff
@@ -18,8 +18,20 @@
 #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #  THE SOFTWARE.
 
+import signal
+import sys
+
 from pbr import version
 
+
+def signal_handler(signal, frame):
+    msg = '{red}Exiting molecule on Ctrl+c.{reset}'
+    print(msg.format(red='\033[31m', reset='\033[0m'))
+    sys.exit()
```

Comment:
So when testing this, I started out using the molecule.utilities helper functions you mentioned, and had this code in cli.py. I noticed there was a window after executing molecule where ^C would still cause a strack trace. Generally this was occurring during the module import process, and (subjectively) was a second or more after executing the program.

So, the motivation for making the change in this way was to place the signal handler as early in the code path as possible, which makes that window smaller. It's still possible to catch it during the pbr import, but it requires a nearly instant ^C to do that. Registering the handler before that import wouldn't pass pep8.

So, ultimately, I don't have strong feelings about where this goes, but the whole point is to catch ^C, and it doesn't do that as effectively if we let it load up all the imports in cli.py and utilities.py before registering the signal handler.